src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -405,4 +411,28 @@ public void EnumerateLinks(Activity activity, int maxLinks)
             }
         }
     }
+
+    private struct ActivitySourceTagsEnumerationState
+    {
+        public SdkLimitOptions SdkLimitOptions;
+
+        public InstrumentationScope InstrumentationScope;
+
+        public void EnumeratTags(ActivitySource activitySource, int maxTags)
+        {
+            var scopeAttributes = this.InstrumentationScope.Attributes;
+
+            foreach (var tag in activitySource.Tags ?? [])
+            {
+                if (this.InstrumentationScope.Attributes.Count() < maxTags)
```

Comment:
Is this `Count()` call here Linq? If `InstrumentationScope.Attributes` is like RepeatablField it should have `.Length` or `.Count` properties to get this more cheaply. IIRC 😄